More portable extraction of short hostname

jasontedor opened the pull request:
This commit increases the portability of extracting the short hostname
on a Unix-like system.

Closes #13107

jasontedor committed:
More portable extraction of short hostname

This commit increases the portability of extracting the short hostname
on a Unix-like system.

Closes #13107

nik9000 commented:
This probably deserves a comment or some helpful person will revert it back to `hostname -s`. Which unices don't have `-s`, btw?

nik9000 commented:
This probably deserves a comment or some helpful person will revert it back to `hostname -s`. Which unices don't have `-s`, btw?

> This probably deserves a comment or some helpful person will revert it back to hostname -s. Which unices don't have -s, btw?

Ah. Solaris 10.

nik9000 commented:
Otherwise LGTM.

jasontedor commented:
@nik9000 Also HP-UX. I've added a comment as you requested. I also discovered in the course of testing this is on Solaris VM that combining the variable definition and export into a single line isn't supported in some shells (namely the Bourne shell which is the default shell on Solaris). I've found a more portable way to do the export as well. I've updated the pull request accordingly.

nik9000 commented:
> I've found a more portable way to do the export as well. I've updated the pull request accordingly.

Bleh. I'm not against just using `#!/bin/bash` instead of `#!/bin/sh`. 

Thanks for adding the comment - that is perfect.

jasontedor commented:
> I'm not against just using `#!/bin/bash` instead of `#!/bin/sh`.

@nik9000 I've been contemplating that as well but at a minimum it is a separate issue from this one.

nik9000 commented:
> @nik9000 I've been contemplating that as well but at a minimum it is a separate issue from this one.


Yeah, probably.
